contrib/agent/src/main/java/io/opencensus/contrib/agent/GrpcContextStrategy.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright 2017, Google Inc.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opencensus.contrib.agent;
+
+import io.grpc.Context;
+import io.opencensus.contrib.agent.bootstrap.ContextStrategy;
+
+/**
+ * Implementation of {@link ContextStrategy} for accessing and manipulating the
+ * {@link io.grpc.Context}.
+ */
+final class GrpcContextStrategy implements ContextStrategy {
```

Comment:
Main reason for the separation is to not ship a copy of the context class with agent (and load it from the bootstrap classloader), but instead use the context class that is loaded from the system classloader (the one that comes with the to-be-instrumented application).

The separation would allow for other implementations of Context, but this is more a side effect of the separation, not the primary intention. I'm not aware of a good example of other (resusable) implementations of a context class.

I've renamed the class to ContextStrategyImpl.